### ISSUE TITLE
roslaunch --files prints error if multiple anon nodes have same id

### ISSUE THREAD

**Issue by hershwg** [ros/ros_comm#186]
When running:

```
roslaunch --files package_name launch_file.launch
```

I get the following error:

```
roslaunch file contains multiple nodes named [/$(anon foo)].
Please check all <node> 'name' attributes to make sure they are unique.
Also check that $(anon id) use different ids.
```

Note that when actually launching the launch file, this will not be a problem because the anonymous node names will be expanded with unique suffixes.  Also, the list of files does not relate to the expansion of anonymous node names, so the file list should be printable regardless.

This is similar in spirit to bugs #94 and #65, both of which suffer because roslaunch simply prints an error and quits when it finds non-unique node names instead of doing as much as possible to help you find your error.

**Comment by hershwg** [comment]
I have a fix for this, by the way, I'm just trying to get the pull request together.

**Comment by hershwg** [comment]
A similar change would fix #65, but in node_args.py, line 85, though it would change the behavior for anonymous node names to show them anonymized.
